### PR TITLE
feat: add article overview page

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary
Closes:
- [[collections] add article overview grid](https://app.clickup.com/t/8678v0j7x)
- [[collections] add article pagination](https://app.clickup.com/t/8678v0j8e) 
- [[collections] add list/grid view toggle](https://app.clickup.com/t/8678v0jbe)
- [[collections] add article overview list view](https://app.clickup.com/t/8678va5g6)

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
some remarks in general:
- can you hook this up to some seeded articles instead of the dummy ones? You can adjust the seeders as needed to add more articles if the current 10 don't suffice (e.g. for pagination)
- pagination is ignored right now, it exists in the UI but it doesn't actually listen to page changes or change in the amount of entries per page

I'm mainly interested in seeing it hooked up to actual article models as that will highlight issues with the components that were based on dummy data

**Comment by ItsANameToo** [comment]
since I ran into it already, UI issue as well:
- the +2 here gets cut off at smaller sizes. This example is at 800px width, even though it seemingly has enough space to fit that badge completely
image

File: resources/js/Components/Articles/ArticleCard/ArticleCard.tsx [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
-export const ArticleCard = ({ article, collections }: Properties): JSX.Element => {
+export const ArticleCard = ({ article }: { article: App.Data.Articles.ArticleData }): JSX.Element => {
     const { t } = useTranslation();
 
     return (
         <a
             data-testid="ArticleCard"
-            href="#"
-            className="group flex w-full max-w-[335px] flex-col overflow-hidden rounded-xl border border-theme-secondary-300 bg-white"
+            href={`/articles/${article.slug}`}
```
Comment (ItsANameToo) [inline resources/js/Components/Articles/ArticleCard/ArticleCard.tsx:12]: add a todo to change this to the route once those are available

File: resources/js/Components/Articles/ArticleCard/ArticleCard.tsx [hunk 2]
```diff
@@ -1,34 +1,30 @@
...
+                    wrapperClassName="h-full [&>span]:h-full"
+                    alt={article.title}
+                    src={article.image}
                 />
             </div>
 
             <div className="flex flex-1 flex-col px-6 py-3">
-                <div className="text-sm font-medium text-theme-secondary-700">24 Oct 2023</div>
+                <div className="text-sm font-medium text-theme-secondary-700">
+                    {DateTime.fromUnix(article.publishedAt).format("D MMM YYYY")}
```
Comment (ItsANameToo) [inline resources/js/Components/Articles/ArticleCard/ArticleCard.tsx:26]: can you move this to a separate component/helper method so it's easy to reuse. It should also use the user's date setting and not hardcoded to this format

File: resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx [hunk 3]
```diff
@@ -1,45 +1,41 @@
...
-export const ArticleListItem = ({ article, collections }: Properties): JSX.Element => {
+export const ArticleListItem = ({ article }: { article: App.Data.Articles.ArticleData }): JSX.Element => {
     const { t } = useTranslation();
 
     return (
         <a
             data-testid="ArticleListItem"
-            href="#"
-            className="flex space-x-3 bg-white p-6 sm:rounded-lg sm:border sm:border-theme-secondary-300"
+            href={`/articles/${article.slug}`}
```
Comment (ItsANameToo) [inline resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx:12]: same todo here

File: resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx [hunk 4]
```diff
@@ -1,45 +1,41 @@
...
             <div className="flex flex-1 flex-col space-y-2">
-                <h4 className="line-clamp-2 text-sm font-medium  leading-[22px] text-theme-secondary-900 sm:text-lg sm:leading-7">
+                <h4 className="line-clamp-2 text-sm font-medium leading-[22px]  text-theme-secondary-900 sm:line-clamp-1 sm:text-lg sm:leading-7">
                     {article.title}
                 </h4>
 
                 <div className="flex items-center space-x-3">
-                    <div className="text-xs font-medium text-theme-secondary-700 sm:text-sm">24 Oct 2023</div>
+                    <div className="text-xs font-medium text-theme-secondary-700 sm:text-sm">
+                        {DateTime.fromUnix(article.publishedAt).format("D MMM YYYY")}
```
Comment (ItsANameToo) [inline resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx:31]: same reuse here

File: resources/js/Components/SortDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import { IconButton } from "@/Components/Buttons";
+import { Dropdown } from "@/Components/Dropdown";
+import { Tooltip } from "@/Components/Tooltip";
+
+interface Properties {
+    disabled?: boolean;
+    children: ReactElement | (({ setOpen }: { setOpen: (open: boolean) => void }) => JSX.Element | JSX.Element[]);
+}
+
+export const SortDropdown = ({ disabled = false, children }: Properties): JSX.Element => {
```
Comment (ItsANameToo) [inline resources/js/Components/SortDropdown.tsx:13]: do we need a new component for this or could we reuse the existing stuff from collections for it?

File: resources/js/Pages/Collections/Components/Articles/ArticlesTab.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                {articlesCount > 0 && displayType === DisplayTypes.List && (
+                    <ArticlesList articles={articles.paginated.data} />
+                )}
+
+                {articlesCount === 0 && query === "" && (
+                    <EmptyBlock>{t("pages.collections.articles.no_articles")}</EmptyBlock>
+                )}
+                {articlesCount === 0 && query !== "" && (
+                    <EmptyBlock>{t("pages.collections.articles.no_articles_with_filters")}</EmptyBlock>
+                )}
```
Comment (ItsANameToo) [inline resources/js/Pages/Collections/Components/Articles/ArticlesTab.tsx:109]: the empty blocks should span the full width, currently they don't
image

File: resources/js/Pages/Collections/View.tsx [hunk 7]
```diff
@@ -169,7 +170,7 @@ const CollectionsView = ({
     const tabChangeHandler = (tab: TabName): void => {
         setSelectedTab(tab);
 
-        setFilterIsDirty(true);
+        // setFilterIsDirty(true);
```
Comment (ItsANameToo) [inline resources/js/Pages/Collections/View.tsx:173]: can you comment on this change for context, I forgot the issue with it

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: resources/js/Pages/Collections/View.tsx (hunk 7 above)
Comment (shahin-hq) [reply resources/js/Pages/Collections/View.tsx:173]: When the tab changes this function set filter is dirty flag true which causes request to backend  and reloads the page with the new parameters which is irrelevant for the articles as I am loading data in the client side - making another request is useless.

File: resources/js/Components/SortDropdown.tsx (hunk 5 above)
Comment (shahin-hq) [reply resources/js/Components/SortDropdown.tsx:13]: Yes, we need that component, I added intentionally to make it reusable: 

Checkout how it is being used in the collections page: https://github.com/ArdentHQ/dashbrd/blob/develop/resources/js/Pages/Collections/Components/CollectionsFilter/CollectionsSorting.tsx

Proposed solution: https://github.com/ArdentHQ/dashbrd/pull/158/files#diff-ee484fe7ed5066defacf979c111321ab58b37534f196df692fbc0d8d8683d3cdR18

Basically, we can refactor the collections page to reuse this component.

File: resources/js/Components/Articles/ArticleCard/ArticleCard.tsx (hunk 1 above)
Comment (shahin-hq) [reply resources/js/Components/Articles/ArticleCard/ArticleCard.tsx:15]: I'll handle it here: https://app.clickup.com/t/8678v0jgg. It is a small change, shouldn't cause too much headache.

File: resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx (hunk 4 above)
Comment (shahin-hq) [reply resources/js/Components/Articles/ArticleListItem/ArticleListItem.tsx]: Should be fine now

File: resources/js/Components/Articles/ArticleCard/ArticleCard.tsx (hunk 2 above)
Comment (shahin-hq) [reply resources/js/Components/Articles/ArticleCard/ArticleCard.tsx]: Should be fine.

**Review by ItsANameToo — APPROVED** [review]
